### ISSUE TITLE
How to get 35.14 BLEU4 score?

### ISSUE THREAD

**Issue by fkjslee** [takase/share_layer_params#6]
dear author, I want to know to get 35.14 BLEU4 score?
I use dataset wmt16 en to de, and run 36 epochs, but only get 19.42 BLEU4 score? (I averaged 32-36 epoch)
below is my run script.
train:
`python -u train.py     data-bin/wmt16_en_de_bpe32k     --arch transformer_wmt_en_de --optimizer adam --adam-betas '(0.9, 0.98)'     --clip-norm 0.0 --lr 0.002 --lr-scheduler inverse_sqrt --warmup-updates 8000     --warmup-init-lr 1e-07 --dropout 0.3 --weight-decay 0.0 --criterion label_smoothed_cross_entropy     --label-smoothing 0.1 --max-tokens 3584 --min-lr 1e-09 --update-freq 32 --log-interval 100     --max-update 50000 --decoder-layers 12 --encoder-layers 12 --share-all-embeddings     --init-type adaptive-profiling  --share-params-cross-layer --share-layer-num 2 --share-type cycle_reverse     --seed 1 --save-dir model-save-dir`
average checkpoint:
`python average_checkpoints.py --inputs model-save-dir --num-epoch-checkpoints 5 --checkpoint-upper-bound 36 --outp
ut model-save-dir/averaged.pt`
generate:
`generate.py data-bin/wmt16_en_de_bpe32k --path model-save-dir/averaged.pt  --beam 4 --lenpen 0.6 --remove-bpe  > generated.result`
I simplly print bleu score by replacing `main(args)` with `print(main(args).result_string())` in line 197 of `generate.py`

**Comment by takase** [comment]
How many GPUs did you use?
As described in ReadMe, we assume that you use 4 GPUs with update-freq 32.

In addition, if you want to achieve high BLEU score such as 35.14 in newstest2014, you need to construct additional training data with back-translation. Please read [our paper](https://arxiv.org/pdf/2104.06022.pdf) for more details.

**Comment by fkjslee** [comment]
3 3090. I didn't expect this. Thx for your reply. I will try again. Thank you

**Comment by fkjslee** [comment]
Did this repo include the code of back-translation? If write back-translation again by myself, maybe I can't get this score.
